### PR TITLE
Render Habit list (disregard branch name)

### PR DESCRIPTION
**Pull request by huachen724**

### CODE REVIEW

File: src/main/java/com/google/sps/data/Habit.java [hunk 1]
```diff
@@ -1,15 +1,18 @@
-package main.java.com.google.sps.data;
+package com.google.sps.data;
```
Comment (changlingh) [inline src/main/java/com/google/sps/data/Habit.java:1]: Small nit: You don't have to change it. But you technically don't have to namespace your code by com.google.sps.

File: src/main/java/com/google/sps/data/HabitList.java [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+
+import com.google.cloud.datastore.TimestampValue;
 
 public final class HabitList {
     
 
-    private final TimeStamp notifyTime;
-    private final string listName;
-    private final string username;
+    //private final TimestampValue notifyTime;
```
Comment (changlingh) [inline src/main/java/com/google/sps/data/HabitList.java:9]: If this is commented out, leave a reason why. 
e.g.
// TODO: Implement notifyTime later.

File: src/main/java/com/google/sps/servlets/CreateHabitServlet.java [hunk 3]
```diff
@@ -15,16 +15,11 @@
...
-        this.habitName = habitName;
-        this.listName = listName;
-    */
-    
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Sanitize user input to remove HTML tags and JavaScript.
     String habit = Jsoup.clean(request.getParameter("habitName"), Whitelist.none());
-    String list = Jsoup.clean(request.getParameter("listName"), Whitelist.none());
+    String list = "Make the bed.";//Jsoup.clean(request.getParameter("listName"), Whitelist.none());
```
Comment (changlingh) [inline src/main/java/com/google/sps/servlets/CreateHabitServlet.java:22]: Nit: start comment on a new line.

File: src/main/java/com/google/sps/servlets/ListHabitServlet.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+// Servlet responsible for listing tasks. 
+@WebServlet("/list-habit")
+public class ListHabitServlet extends HttpServlet {
+    
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    
+    Query<Entity> query = Query.newEntityQueryBuilder().setKind("Habit").setOrderBy(OrderBy.desc("habitName")).build();
```
Comment (changlingh) [inline src/main/java/com/google/sps/servlets/ListHabitServlet.java:30]: I recommend with the repeated use of "habitName" and "Habit" to put it in a class constant.

**Review by changlingh — COMMENTED** [review]
Before you submit, I suggest cleaning up comments sounding unsure of things. e.g. ending in question marks :D

**Review by jessy413 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/DeleteHabitServlet.java [hunk 5]
```diff
@@ -17,10 +17,10 @@ public class DeleteHabitServlet extends HttpServlet {
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-      String habitName = String.parseString(request.getParameter("habitName"));
+      String habitName = request.getParameter("habitName");
   
       Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
-      KeyFactory keyFactory = datastore.newKeyFactory().setKind("Habit");
+      KeyFactory keyFactory = datastore.newKeyFactory().setKind("HabitData");
```
[the remark this answers is not in the document]
Comment (huachen724) [reply src/main/java/com/google/sps/servlets/DeleteHabitServlet.java:23]: just to make the name less confusing!

**Pull request merged**